### PR TITLE
[ADD] module 'account_due_list_days_overdue'

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module adds to the 'Payments and due list' view the number of days that
 an open item is overdue, and up to 5 columns where the amount overdue is
 classified by periods (e.g. 1-30, 31-60,..).

The number of periods/buckets to show in the list and the number of days for
 within each bucket can be defined in the company settings.
# Configuration
- Go to 'Settings / Companies / Companies' and change, in tab 'Aging
  parameters', the number of days overdue for each bucket.
# Usage

To use this module, you need to go to:
- Invoicing / Journal Entries / Payments and due list

image

image

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
Supersedes https://github.com/OCA/account-payment/pull/89

**Comment by JordiBForgeFlow** [comment]
cc @aheficent

File: account_due_list_days_overdue/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2015 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    'name': "Payments Due list days overdue",
+    'version': '8.0.0.1.0',
```
Comment (elicoidal) [inline account_due_list_days_overdue/__openerp__.py:6]: 8.0.1.0.0

File: account_due_list_days_overdue/models/account_move_line.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+from lxml import etree
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    days_overdue = fields.Integer(compute='_compute_days_overdue',
+                                  search='_search_days_overdue',
+                                  string='Days overdue')
+    overdue_term_1 = fields.Float(string='Overdue Term 1',
```
Comment (elicoidal) [inline account_due_list_days_overdue/models/account_move_line.py:15]: I am really fan of this kind of design: I find it not flexible for future use/enhancement. Why not a list (o2m) with limitation to 5 elements?

**Comment by elicoidal** [comment]
@jbeficent Interesting idea but I think you should refactor the design and how to access the payment term fields to make it more generic and flexible for future use

**Comment by JordiBForgeFlow** [comment]
@elicoidal We can cercainly refactor the design. Create a model 'account.days.overdue' that allows the user to enter overdue periods (1-30, 31-60, 61-90, etc..). No overlaps allowed. As records for this model are created, this dynamically creates computed float fields in account.move.line, and the method _compute_overdue_terms calculates the value to put on each new field.

If a user deletes an overdue period, the computed field in account.move.line is also removed.

**Comment by JordiBForgeFlow** [comment]
@elicoidal After banging my head with this approach, I miserably failed. Odoo does not seem to support the option to dynamically create a computed field. It's not possible to dynamically add buckets

**Comment by JordiBForgeFlow** [comment]
Superseded by https://github.com/OCA/account-payment/pull/94. Now closing this one..

**Pull request closed without merging**